transport/httptransport/http_transport.go

```diff
@@ -119,6 +121,16 @@ func (h *httpTransport) Execute(ctx context.Context, transportInfo []byte, dealI
 	}
 	tInfo.URL = u.Url
 
+	// don't allow private network addresses as per https://en.wikipedia.org/wiki/Private_network
```

Comment:
I think we want this to be configurable. We don't really know what kind of setups SPs have and what software they've integrated Boost with... I wouldn't be surprised if some deals are made with private IPs for fetching data.